### PR TITLE
New sphinx #118

### PR DESCRIPTION
**Pull request by nkshaw23**
Documentation now [served](https://onsdigital.github.io/SDG_11.2.1/html/index.html). Surplus folders have been removed, but still couldnt ignore the `html` directory.

Lots of files and additions/deletions in this request, but the majority of them are from generated pages from running the docs. Also lots of commits, but has been a bit of an experiment  and can only check docs have worked when code pushed.

At the moment we have to run `sphinx-build -b html docs/source docs/html` to generate new docs (i.e. when #235 is merged) but would be nice to try and have this run automatically whenever code is pushed.

Note: Also includes code for #240.

### CODE REVIEW

**Comment by nkshaw23** [comment]
Docs updated with new docstrings, and tidied up a couple of other ones.

Can you review geospatial_mods.ward_nrthng_eastng - contains a note that this will be deleted for version 1. Are we deleting now, or removing the note?

**Review by paigeh1 — CHANGES_REQUESTED** [review]
This has been reviewed by myself and James. Do we know if the github pages is being created from all the files in the html folder or is it from the files in the docs folder? If the html folder, then I think the html files should be deleted from the docs folder. Also please see comments that we have left, and give opinions/make changes!

**Comment by Antonio-John** [comment]
closes #194

File: .gitignore [hunk 1]
```diff
@@ -145,8 +145,9 @@ envs
 #data files
 data/
 .SDG_11.2.1
+backup_data
```
[the remark this answers is not in the document]
Comment (nkshaw23) [reply .gitignore]: done

**Comment by nkshaw23** [comment]
Paige's comments have been actioned and docs rebuilt. 

In regards to removing html folder from docs directory, this is where github pages takes the html from so would need to be careful about what, if anything, we remove. A seperate ticket has been created to investigate #253

**Comment by jwestw** [comment]
I am happy that the changes requested have been made, except the one about cleaning up the html folder and a ticket has been created for that. https://github.com/ONSdigital/SDG_11.2.1/issues/253

**Review by jwestw — APPROVED** [review]
Approving as per last comment.

**Review by paigeh1 — APPROVED** [review]
Happy to approve as Nathan had made the changes that James and I requested. See James' comment about tidying html folder as a different ticket has been created. We also may want to think about how this will now work in practice now it has been merged into master (it will come from a different branch, but should work ok).

**Pull request merged**